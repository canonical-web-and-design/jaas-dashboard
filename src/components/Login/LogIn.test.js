import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { mount } from "enzyme";

import LogIn from "./LogIn";

const mockStore = configureStore([]);

describe("LogIn", () => {
  it("renders a login UI if the user is not logged in", () => {
    const store = mockStore({
      root: {}
    });
    const wrapper = mount(
      <Provider store={store}>
        <LogIn>I should not be rendered</LogIn>
      </Provider>
    );
    expect(wrapper.find(".p-button--positive").text()).toBe(
      "Log in to view your models"
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders it's children if the user is logged in", () => {
    const store = mockStore({
      root: {
        controllerConnection: {},
        bakery: {}
      }
    });
    const wrapper = mount(
      <Provider store={store}>
        <LogIn>Hello world!</LogIn>
      </Provider>
    );
    expect(wrapper.find("LogIn").text()).toBe("Hello world!");
    expect(wrapper).toMatchSnapshot();
  });
});
