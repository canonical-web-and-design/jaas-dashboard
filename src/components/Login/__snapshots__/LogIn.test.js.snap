// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn renders a login UI if the user is not logged in 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LogIn>
    <div
      className="login"
    >
      <div
        className="login__inner"
      >
        <img
          alt="JAAS logo"
          className="login__logo"
          src="https://assets.ubuntu.com/v1/a559e2c5-jaas-black-orange-hz-hex.svg"
        />
        <a
          className="p-button--positive"
          rel="noopener noreferrer"
          target="_blank"
        >
          Log in to view your models
        </a>
      </div>
    </div>
  </LogIn>
</Provider>
`;

exports[`LogIn renders it's children if the user is logged in 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LogIn>
    Hello world!
  </LogIn>
</Provider>
`;
